config/config.go

```diff
@@ -88,46 +104,127 @@ var defaultConfig = &config{
 	},
 }
 
+// NewDefaultConfig returns a deep copy of the default configuration
+//
+// By doing this, we ensure that newConfig has its own independent copy of the MeteoProviders slice. Therefore, any changes made to newConfig will not affect defaultConfig.
+func NewDefaultConfig() *config {
+	newConfig := *defaultConfig
+	newConfig.MeteoProviders = make([]MeteoProvider, len(defaultConfig.MeteoProviders))
+	copy(newConfig.MeteoProviders, defaultConfig.MeteoProviders)
+	return &newConfig
+}
+
+// Get returns the current configuration that was last loaded, or loads the configuration if it hasn't been loaded yet
+func Get() (*config, error) {
+	if currentConfig == nil {
+		currentConfig, currentConfigErrors = Load(nil)
+		// return cfg, err
+	}
+	return currentConfig, currentConfigErrors
+}
 
-// New creates a new config object
-func New() (*config, error) {
-	conf := defaultConfig
+// Load loads the configuration from the file if a preferred config is not provided
+//
+// This is useful when Munch is being used as a package and the user wants to dynamically pass their own configuration
```

Comment:
I still don't quite understand why we are passing c in the function, when will this come into play?

Reply:
This is useful, and required, when Munch is used as a package by client programs. Loading Munch config from a file is not practical as the library must be self sufficient. Meaning, a client must be able to programmatically, and dynamically, send in the config for Munch to load it.

Reply:
Ahh i see. Makes sense